docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -294,254 +296,284 @@ TODO
+implement transaction dissemination from lanes. By assigning a P2P channel to each lane, we could
+simply append new transactions to their respective channels and allow the P2P layer to manage the
+order of transmission. We decided against this option mainly because the prioritization logic cannot
+be easily modified without altering the P2P code, potentially affecting other non-mempool channels.
+
+Another drawback is that this option imposes a limit to the number of P2P channels. Channels use a
+byte as an ID, and the current distribution among all reactors goes up to channel `0x61`. For
+example, the current mempool’s P2P channel ID is `0x30`, which would serve as the default lane. We
+could reserve a range of channels for the mempool, such as starting from channel ID `0x80` and above
+(all channels with the most significant byte set to 1). This would provide a maximum of 128 lanes,
+which should suffice for most users.
+
+Nodes would also need to agree on the channel assignments during the P2P handshake. Currently, one
+of the conditions for the handshake to succeed is that there must exist an intersection of P2P
+channels. Since lanes are defined in the application logic, the nodes only need to agree on the
+application version, as it already happens in the current implementation.
+
+### Duality lane/priority
+
+The duality lane/priority could introduce a powerful indirection. The app could just define the lane
+of a transaction in `CheckTx`, but the priority of the lane itself could be configured (and
+fine-tuned) elsewhere. For example, by the app itself or by node operators. The proposed design does
+not support this pattern.
+
+### Custom configuration per lane
+
+A straightforward, future improvement that we leave for after the MVP is to allow customized
+configuration of the lanes instead of sharing the current mempool configuration among lanes. The
+application would need to define new configuration values per lane and pass them to CometBFT during
+the handshake.
+
+### Where to define lanes and priorities
+
+We have considered two alternative approaches for _where_ to configure lanes and priorities:
+1. In `config.toml` or `app.toml`. We have discarded this option as it does not make sense for
+   different nodes to have different lane configurations.
+1. In `ConsensusParams`. There are several disadvantages with this approach. If we allow changing
+   lane information via `ConsensusParams`, the mempool would need to update lanes dynamically. The
+   updating process would be very complex and cumbersome, and not really appealing for an MVP. Two
+   governance proposals would be required to pass to update the lane definitions. A first proposal
+   would be required for upgrading the application, because the lane classification logic (thus the
+   application's code) needs to know the lane configuration beforehand. And a second proposal would
+   be needed for upgrading the lanes via `ConsensusParams`.
```

Comment:
```suggestion
   be needed for upgrading the lanes via `ConsensusParams`.
   While it is true that SDK applications could pass a governance proposal with both elements together,
   it would be something to _always_ do, and it is not clear what the situation would be for non-SDK applications.
```